File: app/src/main/java/com/apkarthik1986/jewelstockmanager/domain/model/ItemStatus.kt
package com.apkarthik1986.jewelstockmanager.domain.model

/**
 * Represents the lifecycle status of a jewelry piece.
 * Weight contribution logic:
 * - AVAILABLE contributes to the box total jewel weight.
 * - SOLD and UNDER_REPAIR are excluded from the active weight total.
 * - UNDER_VALIDATION is treated as active (still in the box, pending confirmation).
 */
enum class ItemStatus(val label: String, val isWeightActive: Boolean) {
    AVAILABLE("Available", true),
    SOLD("Sold", false),
    UNDER_REPAIR("Under Repair", false),
    UNDER_VALIDATION("Under Validation", true);

    companion object {
        fun fromLabel(label: String): ItemStatus =
            entries.firstOrNull { it.label.equals(label, ignoreCase = true) } ?: AVAILABLE
    }
}

File: lib/models/item_status.dart
enum ItemStatus {
  available('Available', true),
  sold('Sold', false),
  underRepair('Under Repair', false),
  underValidation('Under Validation', true);

  const ItemStatus(this.label, this.isWeightActive);
  final String label;
  final bool isWeightActive;

  static ItemStatus fromName(String name) => ItemStatus.values.firstWhere(
        (e) => e.name == name,
        orElse: () => ItemStatus.available,
      );
}
